packages/react-vapor/src/components/flatSelect/FlatSelectSelectors.ts

```diff
@@ -0,0 +1,16 @@
+import {createSelector} from 'reselect';
+import * as _ from 'underscore';
+
+import {IReactVaporState} from '../../ReactVapor';
+import {flatSelectInitialState, IFlatSelectState } from './FlatSelectReducers';
+
+const getInput = (state: IReactVaporState, props: {id: string}): IFlatSelectState => {
+    return _.findWhere(state.flatSelect, {id: props.id}) || flatSelectInitialState;
+};
+
+const getSelectedOptionId = createSelector(getInput, (flatSelect: IFlatSelectState) => flatSelect && flatSelect.selectedOptionId);

```

Comment:
You could add a typing for the return value so that we know what kind of value is returned
```suggestion
const getSelectedOptionId = createSelector(getInput, (flatSelect: IFlatSelectState): string => flatSelect && flatSelect.selectedOptionId);
```